From 0b6541f43804576b5d03091caf6ca1a9c25f5013 Mon Sep 17 00:00:00 2001
From: Joyce Ooi <joyce.ooi@intel.com>
Date: Wed, 7 Oct 2020 12:09:06 +0800
Subject: [PATCH 092/120] Revert "net: eth: altera: fix build errors in QSE
 driver"

commit 99ca8130836badf0ca40514066a425e4534e392f from
https://github.com/altera-opensource/linux-socfpga.git

This reverts commit 4890d1fb915dc61aefd299ff0bc86a73d4aadec7.

Integrated-by: Abhishek Paliwal <paliwal.abhishek@windriver.com>
---
 drivers/net/ethernet/altera/intel_fpga_qse_ll_main.c | 10 ++++------
 1 file changed, 4 insertions(+), 6 deletions(-)

diff --git a/drivers/net/ethernet/altera/intel_fpga_qse_ll_main.c b/drivers/net/ethernet/altera/intel_fpga_qse_ll_main.c
index ce5010b35c1b..8d14b46a1091 100644
--- a/drivers/net/ethernet/altera/intel_fpga_qse_ll_main.c
+++ b/drivers/net/ethernet/altera/intel_fpga_qse_ll_main.c
@@ -1290,7 +1290,7 @@ static void intel_fpga_qse_validate(struct phylink_config *config,
 		   __ETHTOOL_LINK_MODE_MASK_NBITS);
 }
 
-static int intel_fpga_qse_mac_link_state(struct phylink_config *config,
+static void intel_fpga_qse_mac_pcs_get_state(struct phylink_config *config,
 					    struct phylink_link_state *state)
 {
 	struct intel_fpga_qse_private *priv =
@@ -1306,7 +1306,6 @@ static int intel_fpga_qse_mac_link_state(struct phylink_config *config,
 
 	state->link = 1;
 
-	return 0;
 }
 
 static void intel_fpga_qse_mac_an_restart(struct phylink_config *config)
@@ -1372,7 +1371,7 @@ static void intel_fpga_qse_mac_link_up(struct phylink_config *config,
 
 static const struct phylink_mac_ops intel_fpga_qse_phylink_ops = {
 	.validate = intel_fpga_qse_validate,
-	.mac_link_state = intel_fpga_qse_mac_link_state,
+	.mac_pcs_get_state = intel_fpga_qse_mac_pcs_get_state,
 	.mac_an_restart = intel_fpga_qse_mac_an_restart,
 	.mac_config = intel_fpga_qse_mac_config,
 	.mac_link_down = intel_fpga_qse_mac_link_down,
@@ -1612,10 +1611,9 @@ static int intel_fpga_qse_ll_probe(struct platform_device *pdev)
 	spin_lock_init(&priv->ptp_priv.tod_lock);
 
 	/* check if phy-mode is present */
-	priv->phy_iface = of_get_phy_mode(np);
-	if (!priv->phy_iface) {
+	ret = of_get_phy_mode(np, &priv->phy_iface);
+	if (ret) {
 		dev_err(&pdev->dev, "incorrect phy-mode\n");
-		ret = -EINVAL;
 		goto err_free_netdev;
 	}
 
-- 
2.31.1

